graphbolt/src/cnumpy.cc

```diff
@@ -0,0 +1,192 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file cnumpy.cc
+ * @brief Numpy File Fetecher class.
+ */
+
+#include "cnumpy.h"
+
+#include <fcntl.h>
+#include <liburing.h>
+#include <omp.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <regex>
+#include <stdexcept>
+
+#define ALIGNMENT 4096
+
+namespace graphbolt {
+namespace storage {
+
+/**
+ * @brief Read disk numpy file based on given index and transform to tensor.
+ */
+torch::Tensor OnDiskNpyArray::index_select_iouring(torch::Tensor idx) {
+  int feature_fd = open(filename.c_str(), O_RDONLY | O_DIRECT);
+
+  int64_t feature_size = feat_dim * word_size;
+  int64_t num_idx = idx.numel();
+
+  // Multi-thread number.
+  int numthd = 4;
+  omp_set_num_threads(numthd);
+  // Each group size for reader.
+  int64_t group_size = 512;
+
+  char *read_buffer =
+      (char *)aligned_alloc(ALIGNMENT, ALIGNMENT * 2 * group_size * numthd);
+  char *result_buffer =
+      (char *)aligned_alloc(ALIGNMENT, feature_size * num_idx);
+
+  auto idx_data = idx.data_ptr<int64_t>();
```

Comment:
Data dtype could also be `int32`,  add dispatch here.
And also, please add corresponding tests for it.

Reply:
We use unified index data type here https://github.com/dmlc/dgl/pull/7128/files#diff-145c291ca0abd8d19cee18d755fcbe7d9886c5f018fe7528b31b4f37e7459004R28 . Maybe there is no need for type check :)